corehq/apps/users/models.py

```diff
@@ -1764,17 +1764,21 @@ def get_language_code(self):
     def location(self):
         from corehq.apps.locations.models import Location
         if self.location_id:
-            return Location.get(self.location_id)
-        else:
-            return None
+            try:
+                return Location.get(self.location_id)
+            except ResourceNotFound:
+                self.unset_location()
```

Comment:
The problem is that this is called all over the place.  If a user deletes a location, any call to `user.location` will 500.  It doesn't make a lot of sense to do the whole try/except thing in every place this could happen.  I could be talked in to returning `None` if the location isn't found.

AFAIK, there's no way for the user to actually delete a location, although if the location is archived, this should probably act as if the location were deleted (these methods don't filter out archived locations).  Returning `None` would mean that the location association would return if the location is unarchived, but I don't think it's a good idea to have a hidden association.  What if I make these calls return `None` if it's not found, and then call `unset_location` on any relevant users when a location is archived?